### PR TITLE
Add presentation mode prototype

### PR DESCRIPTION
**Pull request by AgapovOne**
Add presentation mode #268

### CODE REVIEW

File: VoiceOver Designer/Documents/WindowManager.swift [hunk 1]
```diff
@@ -69,4 +71,26 @@ extension WindowManager: ProjectRouterDelegate {
         
         recentWindowController.embedProjectsViewControllerInWindow()
     }
+
+    func openPresentationMode(document: NSDocument) {
+        guard let document = document as? VODesignDocument else {
+            return
```
Comment (akaDuality) [inline VoiceOver Designer/Documents/WindowManager.swift:77]: Better to pass VODesignDocument and handle any inconvenience on the upper level

**Review by akaDuality — APPROVED** [review]

File: VoiceOver Designer/Features/Sources/Presentation/PresentationView.swift [hunk 2]
```diff
@@ -0,0 +1,372 @@
...
+    private var scrollContent: some View {
+        if let image = document.image {
+            Image(nsImage: image)
+                .resizable()
+                .frame(
+                    width: document.imageSize.width / document.frameInfo.imageScale,
+                    height: document.imageSize.height / document.frameInfo.imageScale
+                )
+        } else {
+            Color.clear
```
Comment (akaDuality) [inline VoiceOver Designer/Features/Sources/Presentation/PresentationView.swift:72]: `imageSize` will be zero if there is no image

Comment (akaDuality) [reply VoiceOver Designer/Features/Sources/Presentation/PresentationView.swift:72]: Also, potential dividing by zero

**Review by akaDuality — CHANGES_REQUESTED** [review]

File: VoiceOver Designer/Documents/WindowManager.swift (hunk 1 above)
Comment (AgapovOne) [reply VoiceOver Designer/Documents/WindowManager.swift]: Я думал не зависеть от VODesignDocument в протоколе, но в итоге стал зависеть в реализации. И смысл потеряло, да. Спасибо что заметил, исправил

File: VoiceOver Designer/Features/Sources/Presentation/PresentationView.swift (hunk 2 above)
Comment (AgapovOne) [reply VoiceOver Designer/Features/Sources/Presentation/PresentationView.swift:91]: No image is a valid case for any real document?
I see the possiblity in preview, but does it happen in real project?

Comment (akaDuality) [reply VoiceOver Designer/Features/Sources/Presentation/PresentationView.swift:97]: Not yet, but with artboards it will be possible case

Comment (akaDuality) [reply VoiceOver Designer/Features/Sources/Presentation/PresentationView.swift:97]: Not yet, but with artboards it will be possible case

Comment (akaDuality) [reply VoiceOver Designer/Features/Sources/Presentation/PresentationView.swift:97]: You can provide default size here

File: VoiceOver Designer/Features/Sources/Presentation/PresentationView.swift [hunk 3]
```diff
@@ -0,0 +1,399 @@
...
+                    }
+                }
+            }
+        }
+        .padding(.leading, 24)
+        .padding(8)
+        .frame(width: 300, alignment: .leading)
+    }
+
+    private func listButton(_ control: any AccessibilityView, index: Int) -> some View {
```
Comment (AgapovOne) [inline VoiceOver Designer/Features/Sources/Presentation/PresentationView.swift:222]: Caught an issue with indexes and containers here.
@akaDuality a question: is it fine that containers can have elements only 1 level deep?
Isn't it a recursive container that can have containers inside?

Comment (akaDuality) [reply VoiceOver Designer/Features/Sources/Presentation/PresentationView.swift:222]: Yeap, it's app's limitation.

I havn't check support for nested containers in VoiceOver

File: VoiceOver Designer/Features/Sources/Presentation/PresentationView.swift (hunk 2 above)
Comment (AgapovOne) [reply VoiceOver Designer/Features/Sources/Presentation/PresentationView.swift:97]: The only issue is division inside `minimalScaleFactor`, I checked in preview with 0 size.

Fixed it by adding guard in `minimalScaleFactor`

**Comment by AgapovOne** [comment]
Current issues:

- id of AccessibilityView (#272)
- window resize on preview/editor toggle (non-critical)
- right sidebar breaks after one toggle to preview

**Comment by akaDuality** [comment]
I see some constraint conflicts on main branch too, probably it breaks layout after presentation. Will investigate it

**Comment by akaDuality** [comment]
Issue: when I toggle presentation a lot elements placed on canvas multiple times.
Screenshot 2023-10-17 at 19 28 37

If I would draw an element canvas update, but can't click on it and it wouldn't shown in left panel

**Comment by akaDuality** [comment]
I'm going to investigate my issue

**Comment by akaDuality** [comment]
Please, apply this patches in your branch, I've fixed problem with redrawing and responsiveness. 

[0001-Extract-PointerService.patch](https://github.com/VODGroup/VoiceOverDesigner/files/12974092/0001-Extract-PointerService.patch)
[0002-Refactor-subscriptions-subscribe-on-viewDidAppear.patch](https://github.com/VODGroup/VoiceOverDesigner/files/12974093/0002-Refactor-subscriptions-subscribe-on-viewDidAppear.patch)
[0003-Remove-controls-before-drawing.patch](https://github.com/VODGroup/VoiceOverDesigner/files/12974094/0003-Remove-controls-before-drawing.patch)

**Comment by akaDuality** [comment]
As a result, everything should works fine

**Comment by AgapovOne** [comment]
Looks great, bug

> right sidebar breaks after one toggle to preview (probably project controller reference breaks)

is gone. Let's merge?

id issue will be resolved later. Constraints are a different problem with project controller, not with presentation view.

UI for controls in presentation could be larger, but it should be another feature too)

**Comment by AgapovOne** [comment]
Fixed constraints warnings in #275, they are not related to presentation mode

**Comment by AgapovOne** [comment]
Current issue after window refinement: toolbar configuration after toggling presentation mode.

Some toolbar items should not be available for presentation mode, some should be hidden for editor mode.

**Review by akaDuality — APPROVED** [review]

**Comment by akaDuality** [comment]
Congrats!

**Comment by akaDuality** [comment]
Going to finish release with this brand new feature and add to credits on site

**Pull request merged**